src/pages/Categories/Categories.tsx

```diff
@@ -65,7 +48,13 @@ const Categories = () => {
         return (
           <Space size="middle">
             <Button
-              onClick={() => navigate(`/categories/${value.categoryName}`)}
+              onClick={() =>
+                navigate(`/categories/${value.categoryName}`, {
+                  state: {
+                    category: record,
+                  },
+                })
+              }
```

Comment:
Can you please extract this onClick function into separate function as a part of refactoring?